codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoder.java

```diff
@@ -260,7 +260,7 @@ public ChannelFuture writeSettings(ChannelHandlerContext ctx, Http2Settings sett
             ChannelPromise promise) {
         outstandingLocalSettingsQueue.add(settings);
         try {
-            if (connection.isGoAway()) {
+            if (connection.goAwayReceived() || connection.goAwaySent()) {
```

Comment:
I believe the preference here is to be explicit wherever `GOAWAY` is checked.

@Scottmitch @buchgr any thoughts?